### PR TITLE
ci: add release workflow

### PR DESCRIPTION
**Pull request by younesschrifi**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      - name: Log in to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: '$'
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build Docker image
+        run: |
+          docker build -t ghcr.io/${{ github.repository_owner }}/osrd-stable/osrd-images
```
Comment (Khoyo) [inline .github/workflows/release.yml]: You need to tag with the release number here (eg. `ghcr.io/${{ github.repository_owner }}/osrd-stable/osrd-images:v0.3.2`)

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+      - name: Push Docker image
+        run: |
+          docker push ghcr.io/${{ github.repository_owner }}/osrd-stable/osrd-images
```
Comment (Khoyo) [inline .github/workflows/release.yml]: Ditto

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+jobs:
+  build-and-push-docker-image:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
```
Comment (ghost) [inline .github/workflows/release.yml]: Use latest version (v3)

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  release:
+    types: [published]
+
+jobs:
+  build-and-push-docker-image:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
```
Comment (ghost) [inline .github/workflows/release.yml]: Use latest version (v4)

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Log in to GitHub Container Registry
+        uses: docker/login-action@v2
```
Comment (ghost) [inline .github/workflows/release.yml]: Use latest version (v3)

**Review by ghost — CHANGES_REQUESTED** [review]
LGTM overall, we should just use latest version on each action.

**Review by flomonster — APPROVED** [review]

**Review by Khoyo — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**